C++/syntax_test_c.c

```diff
@@ -247,13 +285,329 @@ if (4) {
+
+typedef void (*xhci_get_quirks_t)(struct device *d, struct xhci_hcd *);
+/*<- storage.type.c */
+/*      ^^^^ storage.type.c */
+/*            ^ storage.modifier.c */
+/*             ^^^^^^^^^^^^^^^^^ entity.name.type.c */
+/*                                ^^^^^^ storage.type.c */
+/*                                       ^^^^^^ support.type.c */
+/*                                              ^ storage.modifier.c */
+/*                                               ^ variable.parameter.c */
+/*                                                ^ punctuation.separator */
+/*                                                  ^^^^^^ storage.type.c */
+/*                                                         ^^^^^^^^ support.type.c */
+/*                                                                  ^ storage.modifier.c */
+
+typedef struct Node* (*get_node_t)(struct device *d, struct xhci_hcd *);
+/*<- storage.type.c */
+/*      ^^^^^^ storage.type.c */
+/*             ^^^^ support.type.c */
+/*                 ^ storage.modifier.c */
+/*                    ^ storage.modifier.c */
+/*                     ^^^^^^^^^^ entity.name.type.c */
+/*                                 ^^^^^^ storage.type.c */
+/*                                        ^^^^^^ support.type.c */
+/*                                               ^ storage.modifier.c */
+/*                                                ^ variable.parameter.c */
+/*                                                 ^ punctuation.separator */
+/*                                                   ^^^^^^ storage.type.c */
+/*                                                          ^^^^^^^^ support.type.c */
+/*                                                                   ^ storage.modifier.c */
+
+
+typedef struct Node Node;
+/*<- storage.type.c */
+/*      ^^^^^^storage.type.c */
+/*             ^^^^ support.type.c */
+/*                  ^^^^ entity.name.type.typedef.c */
+typedef struct Node* NodePtr;
+/*<- storage.type.c */
+/*      ^^^^^^storage.type.c */
+/*             ^^^^ support.type.c */
+/*                 ^ storage.modifier.c */
+/*                   ^^^^^^^ entity.name.type.typedef.c */
+
+typedef struct {
+/*<- storage.type.c */
+/*      ^^^^^^storage.type.c */
+/*             ^ meta.struct.body.c punctuation.section.block.begin.c */
+    int data1;
+/*  ^^^ storage.type.c */
+/*      ^^^^^ entity.name.member.indexed.c */
+} newtype;
+/* <- meta.struct.body.c punctuation.section.block.end.c */
+/*^^^^^^^ entity.name.type.typedef.c */
+
+typedef struct MyStruct {
+/*<- storage.type.c */
+/*      ^^^^^^storage.type.c */
+/*             ^^^^^^^^ entity.name.struct.c */
+/*                      ^ meta.struct.body.c punctuation.section.block.begin.c */
+    int data1;
+/*  ^^^ storage.type.c */
+/*      ^^^^^ entity.name.member.indexed.c */
+} newtype;
+/* <- meta.struct.body.c punctuation.section.block.end.c */
+/*^^^^^^^ entity.name.type.typedef.c */
+
+typedef char arrType[6];
```

Comment:
I am somewhat less excited about `storage.modifier` _here,_ though. I understand why you did that, but it looks odd to me, possibly because I'm conditioned to see brackets as punctuation.

Reply:
Seems unconventional indeed (for readers: `[``]` have `storage.modifier` here.

Reply:
I went with `storage.modifier` because it changes the datatype of the variable.

Reading the scope naming:

`storage.modifer`
> Keywords that affect the storage of a variable, function or data structure should use the following scope. 

`meta.brackets`
> Sections of code delineated by square brackets should use the following scope. The `[` and `]` characters should additionally use the punctuation scopes.

It just felt like `storage.modifer` applied for variable declarations. For variable access I don't scope them with `storage.modifer`.